File: com.cssrc.ibms.core.resources/src/main/webapp/jslib/ibms/oa/system/sysObjRights/SysObjRightsService.js
angular.module('SysObjRightsService', ['baseServices'])
.service('SysObjRights', ['$rootScope','BaseService', '$http',function($rootScope,BaseService,$http) {
    var service = {    		
    		//获取SysObjRights的详情
    		detail : function(json,callback){
    			BaseService.post(__ctx +'/oa/system/sysObjRights/getObject.do',json,function(data){
    				if(callback){
	    				callback(data);
	    			 }
    			});
    		},
    		//保存
    		save:function(json,callback){
     			BaseService.post(__ctx +'/oa/system/sysObjRights/save.do',json,function(data){
    				if(callback){
	    				callback(data);
	    			 }
    			});
    		},
    		//根据beanId获取类型列表
    		getRightType:function(beanId,callback){
    			BaseService.post(__ctx +'/oa/system/sysObjRights/getRightType.do',{beanId:beanId},function(data){
    				if(callback){
	    				callback(data);
	    			 }
    			});
    		},
    		//
    		//根据beanId获取类型列表
    		getByObjTypeAndObjectId:function(objType,objectId,callback){
    			BaseService.post(__ctx +'/oa/system/sysObjRights/getObject.do',{objType:objType,objectId:objectId},function(data){
    				if(callback){
	    				callback(data);
	    			 }
    			});
    		}
        }
    return service;
}]);

File: com.cssrc.ibms.core.sysuser/src/main/java/com/cssrc/ibms/core/user/service/SysObjRightsService.java
package com.cssrc.ibms.core.user.service;

import java.util.List;

import javax.annotation.Resource;

import org.springframework.stereotype.Service;

import com.cssrc.ibms.api.core.intf.BaseService;
import com.cssrc.ibms.api.core.util.UniqueIdUtil;
import com.cssrc.ibms.api.sysuser.intf.ISysObjRightsService;
import com.cssrc.ibms.core.db.mybatis.dao.IEntityDao;
import com.cssrc.ibms.core.user.dao.SysObjRightsDao;
import com.cssrc.ibms.core.user.model.SysObjRights;
/**
 * 对象权限service层
 * @author Yangbo 2016-7-22
 *
 */
@Service
public class SysObjRightsService  extends BaseService<SysObjRights> implements ISysObjRightsService
{

	@Resource
	private SysObjRightsDao dao;

	protected IEntityDao<SysObjRights, Long> getEntityDao()
	{
		return this.dao;
	}
	
	/**
	 * 删除某类权限
	 * @param objType
	 * @param objectId
	 */
	public void deleteByObjTypeAndObjectId(String objType, String objectId) {
		this.dao.deleteByObjTypeAndObjectId(objType, objectId);
	}
	
	/**
	 * 保存权限和更新权限
	 * @param sysObjRights
	 */
	public void save(SysObjRights sysObjRights)
	{
		Long id = sysObjRights.getId();
		if ((id == null) || (id.longValue() == 0L)) {
			id = Long.valueOf(UniqueIdUtil.genId());
			sysObjRights.setId(id);
			add(sysObjRights);
		} else {
			update(sysObjRights);
		}
	}
	 /**
	  * 获取某类权限记录
	  * @param objType 权限类别
	  * @param objectId 对应的对象id 多对多关系
	  * @return
	  */
	public List<SysObjRights> getByObjTypeAndObjectId(String objType, String objectId) {
		return this.dao.getObject(objType, objectId);
	}

	public List<SysObjRights> getHashRights(String objType)
	{
/*		Map userMap = this.currentUserService.getUserRelation(ServiceUtil.getCurrentUser());

		Map map = new HashMap();
		map.put("objType", objType);

		map.put("userMap", userMap);

		return this.dao.getBySqlKey("getHashRights", map);*/
		
		return null;
	}
}


